src/include/rados/librados.hpp

```diff
@@ -1003,7 +1003,7 @@ namespace librados
      */
     int aio_operate(const std::string& oid, AioCompletion *c,
 		    ObjectWriteOperation *op, snap_t seq,
-		    std::vector<snap_t>& snaps);
+		    std::vector<snap_t>& snaps, const blkin_trace_info *trace_info = nullptr);
```

Comment:
Could it be fixed by adding an aio_operate_traced function in librados, and calling it from AioObjectRequest if there is trace information, or else defaulting to the regular aio_operate?